port/ztpf/omrosdump_helpers.h

```diff
@@ -0,0 +1,292 @@
+/*******************************************************************************
+ *
+ * (c) Copyright IBM Corp. 1991, 2017
+ *
+ *  This program and the accompanying materials are made available
+ *  under the terms of the Eclipse Public License v1.0 and
+ *  Apache License v2.0 which accompanies this distribution.
+ *
+ *      The Eclipse Public License is available at
+ *      http://www.eclipse.org/legal/epl-v10.html
+ *
+ *      The Apache License v2.0 is available at
+ *      http://www.opensource.org/licenses/apache2.0.php
+ *
+ * Contributors:
+ *    Multiple authors (IBM Corp.) - initial API and implementation and/or initial documentation
+ *    Multiple authors (IBM Corp.) - refactoring and modifications for z/TPF platform
+ *******************************************************************************/
+
+/**
+ * @file
+ * @ingroup Port
+ * @brief z/TPF-specific Dump formatting declarations
+ */
+
+#ifndef _ZTPF_J9OSDUMP_HELPERS_H_INCLUDED
+#define _ZTPF_J9OSDUMP_HELPERS_H_INCLUDED
+
+#include <tpf/i_jdib.h>
+#include <tpf/i_jdbf.h>
+#include <tpf/c_deri.h>
+#include <elf.h>
+#include <errno.h>
+#include <tpf/tpfapi.h>
+
+#ifndef OSDUMP_SYMBOLS_ONLY
+#include "omrport.h"
+#include "portpriv.h" 
+#endif
+
+/*
+ * Get process lock
+ */
+#define PROC_LOCK(proc_lock_ptr, old_key)       \
+    do{                                         \
+        cinfc(CINFC_WRITE, CINFC_CMMFS1);       \
+        old_key = ecbp2()->ce2okey;             \
+        if (is_threaded_ecb()) {                \
+            lockc(proc_lock_ptr,LOCK_O_SPIN); } \
+    } while(0)
+/*
+ * Release process lock
+ */
+#define PROC_UNLOCK(proc_lock_ptr, old_key)     \
+    do{                                         \
+        if (is_threaded_ecb()) {                \
+            unlkc(proc_lock_ptr); }             \
+        ecbp2()->ce2okey = old_key;             \
+        keyrc_okey();                           \
+    } while(0)
+
+/* This data structure is the argument block whose address we need to pass to
+ * ztpfBuildCoreFile() as its one and only parameter. Function
+ * translateInterruptContexts() also uses it since it requires so much of the
+ * same data.
+ *
+ * There are ten (10) data members in this block, it is 0x50 bytes long.
+ */
+typedef struct {
+   siginfo_t				  *sii;			  /* argv[0] = ptr to siginfo_t				*/	

```

Comment:
On a side note, I used the eclipse code formatter, I forget if it added that alignment between type and variable name.